ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala

```diff
@@ -0,0 +1,143 @@
+
+import scala.compat.java8.FutureConverters
+import scala.concurrent.{ExecutionContext, Future}
+
+final class ApiParticipantPruningService private (
+    readBackend: IndexParticipantPruningService with LedgerEndService,
+    writeBackend: WriteParticipantPruningService)(
+    implicit grpcExecutionContext: ExecutionContext,
+    logCtx: LoggingContext)
+    extends ParticipantPruningServiceGrpc.ParticipantPruningService
+    with GrpcApiService {
+
+  private val logger = ContextualizedLogger.get(this.getClass)
+
+  override def bindService(): ServerServiceDefinition =
+    ParticipantPruningServiceGrpc.bindService(this, DirectExecutionContext)
+
+  override def prune(request: PruneRequest): Future[PruneResponse] = {
+    val submissionIdOrErr = SubmissionId
+      .fromString(
+        if (request.submissionId.nonEmpty) request.submissionId else UUID.randomUUID().toString)
+      .left
+      .map(err => ErrorFactories.invalidArgument(s"submission_id $err"))
+
+    submissionIdOrErr.fold(
+      Future.failed,
+      submissionId =>
+        LoggingContext.withEnrichedLoggingContext("submissionId" -> submissionId) {
+          implicit logCtx =>
+            (for {
+
+              pruneUpTo <- validateRequest(request: PruneRequest)
+
+              // If write service pruning succeeds but ledger api server index pruning fails, the user can bring the
+              // systems back in sync by reissuing the prune request at the currently specified or later offset.
+              _ <- pruneWriteService(pruneUpTo, submissionId)
+
+              pruneResponse <- pruneLedgerApiServerIndex(pruneUpTo)
+
+            } yield pruneResponse).andThen(logger.logErrorsOnCall[PruneResponse])
+      }
+    )
+  }
+
+  private def validateRequest(request: PruneRequest)(
+      implicit logCtx: LoggingContext): Future[Offset] = {
+    (for {
+      pruneUpToString <- checkOffsetIsSpecified(request.pruneUpTo)
+      pruneUpTo <- checkOffsetIsHexadecimal(pruneUpToString)
+    } yield (pruneUpTo, pruneUpToString))
+      .fold(Future.failed, o => checkOffsetIsBeforeLedgerEnd(o._1, o._2))
+  }
+
+  private def pruneWriteService(pruneUpTo: Offset, submissionId: SubmissionId)(
+      implicit logCtx: LoggingContext): Future[Unit] = {
+    logger.info(
+      s"About to prune participant ledger up to ${pruneUpTo} inclusively starting with the write service")
+    FutureConverters
+      .toScala(writeBackend.prune(pruneUpTo, submissionId))
+      .flatMap {
+        case PruningResult.NotPruned(status) => Future.failed(ErrorFactories.grpcError(status))
+        case PruningResult.ParticipantPruned =>
+          logger.info(s"Pruned participant ledger up to ${pruneUpTo} inclusively.")
+          Future.successful(())
+      }
+  }
+
+  private def pruneLedgerApiServerIndex(pruneUpTo: Offset)(
+      implicit logCtx: LoggingContext): Future[PruneResponse] = {
+    logger.info(s"About to prune ledger api server index to to ${pruneUpTo} inclusively")
```

Comment:
```suggestion
    logger.info(s"About to prune ledger api server index to ${pruneUpTo} inclusively")
```

Reply:
Please `import ApiOffset.ApiOffsetSConverter` and use `${pruneUpTo.toApiString}` in the log messages to make the offset somewhat readable.